### PR TITLE
[Backport 2.x][Maintenance] Bumps `re2` and `supertest`

### PR DESCRIPTION
**Pull request by ananzh**
### Description
* Removes manual resolution on `qs`. The latest version of `supertest` depends on v6.11.0.
* Adds steps for upgrading `re2`.
* Addresses CVE-2022-24999 (no issue opened)

### Backport PR:
https://github.com/opensearch-project/OpenSearch-Dashboards/pull/3018

 
### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -194,7 +193,7 @@
     "pegjs": "0.10.0",
     "proxy-from-env": "1.0.0",
     "query-string": "^6.13.2",
-    "re2": "^1.15.4",
+    "re2": "1.17.4",
```
Comment (joshuarrrr) [inline package.json:196]: we don't want `^1.17.4`?

File: yarn.lock [hunk 2]
```diff
@@ -3569,9 +3569,9 @@
     csstype "^3.0.2"
 
 "@types/superagent@*":
-  version "4.1.15"
```
Comment (joshuarrrr) [inline yarn.lock:3572]: did we run `yarn bootstrap` to automatically generate these changes?

**Review by joshuarrrr — APPROVED** [review]
A couple questions, but not blocking

**Review by manasvinibs — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ananzh) [reply package.json:196]: I have the same question... but it is better to be aligned with main.

File: yarn.lock (hunk 2 above)
Comment (ananzh) [reply yarn.lock:3572]: Yes. I run `yarn bootstrap` and yarn.lock has some auto updates

**Pull request merged**